airflow/timetables/interval.py

```diff
@@ -158,7 +158,7 @@ def __eq__(self, other: Any) -> bool:
 
         This is only for testing purposes and should not be relied on otherwise.
         """
-        if not isinstance(other, CronDataIntervalTimetable):
+        if not isinstance(other, _CronMixin):
```

Comment:
This should be something like `isinstance(other, type(self)`, otherwise a `CronDataIntervalTimetable` and a `CronTriggerTimetable` may compare to True, which is not what we want.

Reply:
Thanks. Fixed by https://github.com/apache/airflow/pull/23662/commits/95dbb4ef3b52c9447636a4b5b3b3c95e9e9d731b